src/OpenTelemetry/Trace/BatchExportActivityProcessor.cs

```diff
@@ -165,5 +289,32 @@ protected override void Dispose(bool disposing)
                 this.disposed = true;
             }
         }
+
+        private void ExporterProc()
+        {
+            var triggers = new[] { this.exportTrigger, this.shutdownTrigger };
+
+            while (true)
+            {
+                // only wait when the queue doesn't have enough items, otherwise keep busy and send data continuously
+                if (this.circularBuffer.Count < this.maxExportBatchSize)
+                {
+                    WaitHandle.WaitAny(triggers, this.scheduledDelayMillis);
+                }
+
+                if (this.circularBuffer.Count > 0)
+                {
+                    this.exporter.Export(new Batch<Activity>(this.circularBuffer, this.maxExportBatchSize));
+
+                    this.dataExportedNotification.Set();
+                    this.dataExportedNotification.Reset();
```

Comment:
I think we also need a `this.exportTrigger.Reset();` here or it's going to be stuck in the signaled state. Alternatively, AutoResetEvent should work for `exportTrigger`. I'm pretty sure it stays signaled until at least one thing is triggered, which in our case would be an export.

Reply:
Nice catch, that line was missed while I was cleaning up the code.